kensu/requests/api.py

```diff
@@ -11,33 +13,49 @@ def wrapper(*args, **kwargs):
 
         kensu = KensuProvider().instance()
 
-        result_pk = DataSourcePK(location=result.url,
-                                 physical_location_ref=kensu.default_physical_location_ref)
-        result_ds = DataSource(name=result.url, format='API',
-                               pk=result_pk)._report()
+        try:
+            fields_dict = flatten(result.json())
 
+            # Construct the file location
 
-        fields_dict = flatten(result.json())
+            url = kwargs['url'] if 'url' in kwargs else args[0]
+            try:
+                params = kwargs['params'] if 'params' in kwargs else args[1]
```

Comment:
hmm, but what if `args[1]` is not params, e.g. no params passed at all...?
may also help to check type of `params` to be as expected - a `dict`

Reply:
yep then `params = None` in the `except`

Reply:
i mean in theory we can get any other rubish in `args[1]`, isn't that possible ? unless we checked all usages of it